_posts/2019-04-05-gluster-3-12-vs-6-a-performance-oriented-overview.md

```diff
@@ -0,0 +1,340 @@
+---
+layout: post
+title: Gluster 3.12 vs 6 - A performance oriented overview
+author: ezio-auditore 
+---
+**Gluster 3.12 vs 6 - A performance oriented overview**
+
+With the release of gluster 6 I decided to have a performance benchmark test with gluster 3.12 version to see if there was really any improvement seen. Used [pg-bench](https://www.postgresql.org/docs/10/pgbench.html) tool to measure the performance. 
+ 
+PG-Bench is a simple program for running benchmark tests on PostgreSQL. It runs the same sequence of SQL commands over and over, possibly in multiple concurrent database sessions, and then calculates the average transaction rate (transactions per second). By default, pgbench tests a scenario that is loosely based on TPC-B, involving five `SELECT`, `UPDATE`, and `INSERT` commands per transaction
+
+The methods I used to measure them are very simple. For those who want to know stick around others can jump straight to the results.
+
+I started with installing gluster 3.12 hyper-converged  environment on three bare metal machines using the cockpit-ovirt dashboard. I continued with installing hosted-engine managing the cluster. After the initial setup was complete I created 2 vms, one having disk allocation policy as "Preallocated" and the other as "Thin provisioned".Once the vms were up it was time to setup pgbench.
+
+**Bare metal machine specs** 
+
+OS  : Red Hat Enterprise Linux Server release 7.7 Beta (Maipo)
+
+Processor: 	Intel(R) Xeon(R) CPU E5-1620 v4 @ 3.50GHz
+
+Execution Technology : 4 cores; 8 threads
+
+**Storage specs**
+
+Model	Smart HBA H240
+
+Firmware Version	4.52
+
+Controller Type	HPE Smart Array
+
+Media Type	SSD
+
+Capacity	400 GB
+
+Following are the steps I did for setting pg bench :-
+
+
+
+1. yum install [https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm](https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm)
```

Comment:
these two steps can be changed as code

````
```
$ yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
$ yum install postgresql-contrib postgresql-server
```
````